### PR TITLE
Pt 166233685 fate oracles

### PR DESCRIPTION
**Pull request by gorillainduction**
https://www.pivotaltracker.com/story/show/166233685

### CODE REVIEW

File: apps/aecontract/test/aecontract_SUITE.erl [hunk 1]
```diff
@@ -1750,7 +1775,10 @@ sophia_oracles(_Cfg) ->
     Ct1 = ?call(create_contract, Acc, oracles, {}, #{amount => 1000000}),
     QuestionType = {variant_t, [{why, [word]}, {how, [string]}]},
     AnswerType   = {variant_t, [{noAnswer, []}, {yesAnswer, [QuestionType, string, word]}]},
-    Question1    = {variant, 1, [<<"birds fly?">>]},
+    Question1    = case ?IS_FATE_SOPHIA(vm_version()) of
```
Comment (hanssv) [inline apps/aecontract/test/aecontract_SUITE.erl]: Maybe use `IF_AEVM`?

File: apps/aecontract/test/aecontract_SUITE.erl [hunk 2]
```diff
@@ -143,8 +143,19 @@
 -define(MINER_PUBKEY, <<12345:?MINER_PUB_BYTES/unit:8>>).
 -define(BENEFICIARY_PUBKEY, <<12345:?BENEFICIARY_PUB_BYTES/unit:8>>).
 
--define(CHAIN_RELATIVE_TTL_MEMORY_ENCODING(X), {variant, 0, [X]}).
--define(CHAIN_ABSOLUTE_TTL_MEMORY_ENCODING(X), {variant, 1, [X]}).
+-define(CHAIN_RELATIVE_TTL_MEMORY_ENCODING_TYPE(X), tuple()).
+-define(CHAIN_ABSOLUTE_TTL_MEMORY_ENCODING_TYPE(X), tuple()).
+
+-define(CHAIN_RELATIVE_TTL_MEMORY_ENCODING(X),
+        case ?IS_FATE_SOPHIA(vm_version()) of
```
Comment (hanssv) [inline apps/aecontract/test/aecontract_SUITE.erl]: Shorter with `?IF_AEVM`

File: apps/aecontract/test/aecontract_SUITE.erl [hunk 3]
```diff
@@ -1777,11 +1807,13 @@ sophia_oracles_type_error_out(_Cfg) ->
     Ct = <<CtId:256>> = ?call(create_contract, Acc, oracles, {}, #{amount => 100000}),
     QueryFee          = 100,
     TTL               = 15,
-    CtId              = ?call(call_contract, Acc, Ct, registerStringStringOracle, word, {CtId, QueryFee, FixedTTL(TTL)}),
+    Oracle            = ?call(call_contract, Acc, Ct, registerStringStringOracle, word, {Ct, QueryFee, FixedTTL(TTL)}),
+    ?assertMatchAEVM(CtId, Oracle),
```
Comment (hanssv) [inline apps/aecontract/test/aecontract_SUITE.erl]: Just `?assertMatchVM` perhaps?

**Comment by gorillainduction** [comment]
@hanssv @UlfNorell I changed the base branch to a temporary master branch.

**Pull request merged**